File: 929. Unique Email Addresses.py
class Solution:
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        email_set = set()
        for email in emails:
            local_name,domain_name = email.split("@")
            local_name=local_name.split('+')[0]
            local_name ="".join(local_name.split('.'))      
            email = local_name +'@' + domain_name
            email_set.add(email)
        return len(email_set)
            

File: 929. Unique Email Addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
    unordered_set<string> clearedEmails;
    for (auto& email:emails){
        string local=email.substr(0,email.find('@'));
        string domain =email.substr(email.find('@'));
        local=local.substr(0,email.find('+'));
        string newLocal="";
        for (auto c:local){
            if (c!='.') newLocal+=c;
        }

        clearedEmails.insert(newLocal+domain);        
     }

    return clearedEmails.size();
    }
};
